### PR TITLE
Basic Marathon client library

### PR DESCRIPTION
**Pull request by jerith**

### CODE REVIEW

File: test/test_helper.exs [hunk 1]
```diff
@@ -1 +1,3 @@
+# Code.require_file(Path.join([__DIR__, "marathon_client", "marathon_client_helper.exs"]))
```
Comment (JayH5) [inline test/test_helper.exs]: What's this?

File: mix.lock [hunk 2]
```diff
@@ -1,11 +1,25 @@
 %{
+  "certifi": {:hex, :certifi, "2.0.0", "a0c0e475107135f76b8c1d5bc7efb33cd3815cb3cf3dea7aefdd174dabead064", [:rebar3], [], "hexpm"},
   "chatterbox": {:git, "https://github.com/tony612/chatterbox.git", "c7c2ab4c3e74618cfa78fdbf6444fac3952d194b", [branch: "my-fix"]},
+  "conform": {:hex, :conform, "2.5.2", "7035787a9c09d28607745444e7a1700426dc47c452634a5694033fa2fbb3414c", [:mix], [{:neotoma, "~> 1.7.3", [hex: :neotoma, repo: "hexpm", optional: false]}], "hexpm"},
   "cowboy": {:hex, :cowboy, "2.2.2", "6cde634da1da8b1d17dd2b20f5f6f91ab8ac7fa1b5106d13dad63c905ab76004", [], [{:cowlib, "~> 2.1.0", [hex: :cowlib, repo: "hexpm", optional: false]}, {:ranch, "~> 1.4.0", [hex: :ranch, repo: "hexpm", optional: false]}], "hexpm"},
   "cowlib": {:hex, :cowlib, "2.1.0", "f73658b93dd043af40400c3e4fd997068ebd0c617f8c8f4cd003a1a78ebf94f5", [], [], "hexpm"},
+  "distillery": {:hex, :distillery, "1.5.2", "eec18b2d37b55b0bcb670cf2bcf64228ed38ce8b046bb30a9b636a6f5a4c0080", [:mix], [], "hexpm"},
```
Comment (JayH5) [inline mix.lock]: Curious as to what is pulling in distillery?

File: test/test_helper.exs (hunk 1 above)
Comment (jerith) [reply test/test_helper.exs]: A leftover thing that I forgot to delete after I moved it to some of the `marathon_client` test modules.

`ExUnit`'s test runner does something a lot like that to load `test/test_helper.exs` and make its contents available to the rest of the test modules. Unlike `.ex` files (which are all compiled and loaded upfront), `.exs` files are interpreted and only loaded when something explicitly loads them. I didn't want to have all the marathon/sse test helpers (and various other component-specific bits when we get there) dumped into a giant global `test_helper.exs`, so I put them in their own file and load them only where they're actually needed.

File: mix.lock (hunk 2 above)
Comment (jerith) [reply mix.lock]: Probably `sse_test_server`. It's tagged `runtime: false`, but that may not be enough.

**Pull request merged**